create_auction_result.php

```diff
@@ -1,33 +1,144 @@
+        // If reserve price is provided but not a positive number, show an error
+        $errors[] = "Reserve price must be a positive number.";
+    } elseif ($reservePrice < $startPrice) {
+        // If reserve price is lower than start price, show an error
+        $errors[] = "Reserve price cannot be lower than the start price.";
+    }
+
+    // Validate end date - you may want to check if the date is in the future
+    if (empty($endDate) || strtotime($endDate) <= time()) {
+        $errors[] = "Invalid end date.";
+    }
+
+    // If there are validation errors, display them
+    if (!empty($errors)) {
+        echo('<div class="text-center text-danger">Error creating auction. Please fix the following issues:<br>');
+        foreach ($errors as $error) {
+            echo('- ' . $error . '<br>');
+        }
+        echo('</div>');
+    } else {
 
-// If all is successful, let user know.
-echo('<div class="text-center">Auction successfully created! <a href="FIXME">View your new listing.</a></div>');
+        // If everything looks good, make the appropriate call to insert data into the database. */
+        $query = "INSERT INTO Auction (title, description, seller_id, start_price, reserved_price, current_price, end_time) 
+                VALUES ('$title', '$details', $seller_id,  $startPrice, $reservePrice, $startPrice, '$formattedEndDate')";
 
+        // Execute the query
+        $result = mysqli_query($connection, $query);
+        // $connection->query($query); 
+
+        if ($result) {
+            // If the insertion was successful, get the auction ID
+            $auction_id = mysqli_insert_id($connection);
+
+            // Now, add rows to the Auction_Product table
+            foreach ($_POST['selectedItems'] as $item_id) {
+                $insertProductQuery = "INSERT INTO Auction_Product (item_id, auction_id) VALUES ('$item_id', '$auction_id')";
+                $insertProductResult = mysqli_query($connection, $insertProductQuery);
+
+                if (!$insertProductResult) {
+                    // Handle error if any
+                    echo('<div class="text-center text-danger">Error adding items to the Auction Product table. Please try again.</div>');
+                    echo "error: " . mysqli_error($connection);
+                    exit();
+                }
+            }
+            // Notify the user that the auction and associated items were added successfully
+            echo('<div class="text-center">Auction and associated items successfully created! <a href="FIXME">View your new listing.</a></div>');
+
+            //Update item availability in the Item table after it has successfully been added to the Auction and Auction_Product table
+            foreach ($_POST['selectedItems'] as $itemId) {
+                $updateQuery = "UPDATE Item SET is_available = 0 WHERE id = $itemId";
+                $updateResult = mysqli_query($connection, $updateQuery);
+            

```

Comment:
Sorry @jasmehr i think we also need the assertion for `if (current session id !== item.user_id)` so it can be passed by api call

Reply:
my mistake, adding another commit rn.